### PR TITLE
Use x64 binaries when project's targeting AnyCPU

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes https://github.com/aspnet/MvcPrecompilation/issues/240

### CODE REVIEW

File: test/FunctionalTests/FunctionalTests.csproj [hunk 1]
```diff
@@ -33,7 +33,6 @@
   </ItemGroup>
 
   <ItemGroup>
-    <ProjectReference Include="..\..\testapps\ApplicationUsingPrecompiledViewClassLibrary\ApplicationUsingPrecompiledViewClassLibrary.csproj" />
```
Comment (pranavkm) [inline test/FunctionalTests/FunctionalTests.csproj]: Missed removing this when I deleted the project in an earlier commit.

File: test/FunctionalTests/DesktopTests/SimpleAppTest_WithAnyCPU_Desktop.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+using Microsoft.AspNetCore.Server.IntegrationTesting;
+using Microsoft.AspNetCore.Testing.xunit;
+using Microsoft.Extensions.Logging.Testing;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace FunctionalTests
+{
+    [OSSkipCondition(OperatingSystems.Linux)]
+    [OSSkipCondition(OperatingSystems.MacOSX)]
```
Comment (ajaybhargavb) [inline test/FunctionalTests/DesktopTests/SimpleAppTest_WithAnyCPU_Desktop.cs]: [OSSkipCondition(OperatingSystems.Linux | OperatingSystems.MacOSX)]

File: src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation/build/netstandard2.0/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.targets [hunk 3]
```diff
@@ -85,8 +85,8 @@
     Condition="'$(MvcRazorCompileOnPublish)'=='true' AND '$(_MvcViewCompilationAddDesktopReferences)'!='false' AND '$(TargetFrameworkIdentifier)'=='.NETFramework'">
 
     <PropertyGroup Condition="'$(_MvcViewCompilationBinaryPath)'==''">
-      <_MvcViewCompilationBinaryPath Condition="'$(PlatformTarget)'=='x64'">$(MSBuildThisFileDirectory)$(MSBuildThisFileName)-x64.exe</_MvcViewCompilationBinaryPath>
```
Comment (ajaybhargavb) [inline src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation/build/netstandard2.0/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.targets]: Would be nice to have a comment here saying we prefer x64 bits by default. It wasn't obvious to me.

File: test/FunctionalTests/FunctionalTests.csproj [hunk 4]
```diff
@@ -1,8 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>netcoreapp2.1</TargetFrameworks>
-    <TargetFrameworks Condition="'$(DeveloperBuild)'!='true'">$(TargetFrameworks);netcoreapp2.0</TargetFrameworks>
+    <TargetFrameworks>$(StandardTestTfms)</TargetFrameworks>
```
Comment (pranavkm) [inline test/FunctionalTests/FunctionalTests.csproj]: @ajaybhargavb fixing up the csproj so we actually run desktop tests.

**Pull request merged**